core/src/impl/Kokkos_Default_GraphNode_Impl.hpp

```diff
@@ -92,6 +92,18 @@ struct GraphNodeBackendSpecificDetails {
     m_is_aggregate = true;
   }
 
+  // A node is awaitable if it can execute a kernel.
+  // A root node or an aggregate node cannot be waited for, because it does
+  // not launch anything.
+  bool awaitable() const { return (!m_is_root) && (!m_is_aggregate); }
```

Comment:
Is this a new publicly visible API?

Reply:
It's a method of `Kokkos::Impl::GraphNodeBackendSpecificDetails` so I guess no one (that is sufficiently careful) will use it.

However, I guess I can put that `awaitable` method within the `private` methods if you want.

Reply:
It is fine as proposed